http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/Http2GoAwayEventListener.java

```diff
@@ -41,7 +42,7 @@ public Http2GoAwayEventListener(Channel parentChannel) {
     @Override
     public void onGoAwayReceived(int lastStreamId, long errorCode, ByteBuf debugData) {
         Http2MultiplexedChannelPool channelPool = parentChannel.attr(ChannelAttributeKey.HTTP2_MULTIPLEXED_CHANNEL_POOL).get();
-        GoAwayException exception = new GoAwayException(errorCode, debugData.retain());
+        GoAwayException exception = new GoAwayException(errorCode, debugData.toString(StandardCharsets.UTF_8));
```

Comment:
Should we release the ByteBuf after this? Preferably in a finally block?

Reply:
I do not believe so... The original bug was that by retaining the debugData, the refCount was increased, but never released.  I restructured the code, to convert to a String and pass into the GoAwayException, so we always know the lifecycle of that bytebuf, and it seemed tidier than try/final the modifying of the refCount for processing that happens all within this method..

The Javadoc for https://netty.io/4.1/api/io/netty/handler/codec/http2/DefaultHttp2Connection.html goAwayReceived says 
> debugData - The Additional Debug Data in the GOAWAY frame. Note that reference count ownership belongs to the caller (ownership is not transferred to this method).
So I think it's safe to just use the debugData content within the scope of the method, and the code shouldn't release it.

I will do a quick check, to see if this bug appears elsewhere.

Reply:
I will update the changelog once we've run the fix for a week, and I'm confident that it fixes our problem :)

Reply:
Looks like the fix, fixes the problem

Reply:
Nice, thanks for verifying.